pydtk/io/writer.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright Toolkit Authors
+
+"""pydtk modules."""
+
+from abc import ABCMeta
+
+from pydtk.models import MODELS_BY_PRIORITY, MetaDataModel
+from pydtk.preprocesses import PassThrough
+
+
+class NoModelMatchedError(BaseException):
+    """Error of module matching."""
+
+    pass
+
+
+class BaseFileWriter(metaclass=ABCMeta):
+    """Base file writer."""
+
+    _model = None   # model used for loading a file
+
+    def __init__(self, **kwargs):
+        self.preprocesses = [PassThrough()]
+
+    @classmethod
+    def _select_model(cls, file_metadata):
+        """Select a proper model based on the given file-metadata.
+
+        Args:
+            file_metadata (object): an MetaDataModel object
+
+        """
+        priorities = MODELS_BY_PRIORITY.keys()
+        for priority in sorted(priorities, reverse=True):
+            for model in MODELS_BY_PRIORITY[priority]:
+                if model.is_loadable(**file_metadata.data):
+                    return model
+        raise NoModelMatchedError('No suitable model found for loading data: {}'.
+                                  format(file_metadata))
+
+    @property
+    def model(self):
+        """Return model."""
+        return self._model
+
+    @model.setter
+    def model(self, model):
+        self._model = model
+
+    def add_preprocess(self, preprocess):
```

Comment:
I think this method is not necessary as BaseFileWriter does not do any pre-processes.